usagov/usagov-2021
# Usagov 817 extend error checking in tome

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-817

## Description

Extend the error checking in scripts/tome-sync.sh to identify this problem and not publish when it happens. It checks for links to /node/[some number] anywhere on the home page, not just in the mobile menu and emits a unique warning we can search for.
I also added two more things to the grep that checks for spanish and english content.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Run `tome-run.sh` if you don't have the static site files in your local environment.
2. Find a folder called `html` and check that you have these files:
      * `html/index.html`
      * `html/es/index.html`
3. Open Docker Desktop and go to the 'Containers' tab.
4. Locate the `cms` inside the usagov-2021 container
5. Click the 3 dots on the right side and select the option `Open in terminal`
<img width="1206" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/0e405432-ec04-40cd-bdff-e51f6831ccc3">

6. Type `./scripts/tome-sync.sh` in the terminal
7. Once finished, click Control + F and search for "index.html menu appears to contain links that goes to /node/ pages" to verify that you haven't received any of these warnings.
8. Go to VSCode and open this file `html/index.html` (Make sure to leave it open because it's going to be deleted when tome-sync.sh finishes and there's an error.)
9. Find the menu and change one of the links to `/node/[some number]`
<img width="628" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/ba386a26-c1d5-4d64-94f7-29733ecfc31c">
10. Open this file `html/es/index.html` (Make sure to leave it open because it's going to be deleted when tome-sync.sh finishes and there's an error.)
11. Find the menu and change one of the links to `/node/[some number]`
<img width="792" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/552bce28-e979-4185-8715-0d927d3293e6">

12. Run the tome-sync.sh script and you should see these two warnings in the terminal
<img width="667" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/49695932-313b-48fd-8573-9d60a924fd46">

13. Open this file `html/index.html` (Make sure to leave it open because it's going to be deleted when tome-sync.sh finishes and there's an error.)
14. Find the menu and change the content in one of the links to one of these: `Todos los temas y servicios or Acerca de USAGov en Español or Fuerzas Armadas de EE. UU. y veteranos`
<img width="791" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/6f9809d7-7189-4f43-a1cd-02dad61bd691">

15. Open this file `html/es/index.html` (Make sure to leave it open because it's going to be deleted when tome-sync.sh finishes and there's an error.)
16. Find the menu and change the content in one of the links to one of these: `All topics and services or About USAGov or Military and veterans`
<img width="757" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/5b6b69a3-ea4c-4bb0-916d-7acd55f63417">

17. Run the tome-sync.sh script and you should see these two warnings in the terminal
<img width="489" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/5200e5f5-3198-46db-be98-43a927785b93">

### Requires New Config

### Requires New Content

### Validation Steps

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
12. Find the commit of this pull request.
14. Build and deploy the changes.
15. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
